src/ledger/database/hashmap.zig

```diff
@@ -442,5 +455,7 @@ const SharedHashMap = struct {
 };
 
 comptime {
-    _ = &database.interface.testDatabase(SharedHashMapDB);
+    if (build_options.blockstore_db == .hashmap) {
```

Comment:
When testing, we want to test all implementations. Ditto for the other 2 places where you do this.

Reply:
I made the hashmap tests unconditional, since they don't have any external dependencies: a283dcce07d1617aabf46810a1329ed753a9f247. But I haven't done it for rocksdb or lmdb.

Running all the tests was my initial intent, but I changed my mind because it's not actually possible to run all the tests in a single build.

Sure, you could run the tests in testDatabase for each database. But you can't run all the ledger tests for every database, unless you rebuild it separately for each database and execute separate test runs.

If you want to truly properly test all the databases, there is no avoiding it. You need to run three separate commands. I added these to CI to ensure they all pass before a merge.
```
zig build test
zig build test -Dblockstore-db=lmdb -Dfilter=ledger
zig build test -Dblockstore-db=hashmapdb -Dfilter=ledger
```

At that point, why force people to build every database when running tests? Sure it gets a little more test coverage, but it's still not sufficient, and it wastes time building extra dependencies.

Reply:
Hmm, I didn't consider that the ledger tests. 

I'm convinced, the CI checks should be sufficient, and it's not very likely that there would be a bug coming from a specific backend.